src/aat/resources/features/F-037/F-037.feature

```diff
@@ -0,0 +1,66 @@
+@F-037
+Feature: F-037: Create event (V2)
+
+  Background: Load test data for the scenario
+    Given an appropriate test context as detailed in the test data source
+
+  @S-024
+  Scenario: should create event successfully for correct inputs
+    Given a user with [an active profile in CCD]
+    And a case that has just been created as in [Standard_Full_Case_Creation_Data]
+    And a successful call [to get an event token for just created case] as in [S-024-Prerequisite]
+    When a request is prepared with appropriate values
+    And it is submitted to call the [Create event (V2)] operation of [CCD Data Store]
+    Then a positive response is received
+    And the response [has the 200 return code]
+    And the response has all other details as expected
+
+  @S-022
+  Scenario: must return negative response when request does not provide valid authentication credentials
+    Given a user with [an active profile in CCD]
+    When a request is prepared with appropriate values
+    And the request [does not provide valid authentication credentials]
+    And it is submitted to call the [Create event (V2)] operation of [CCD Data Store]
+    Then a negative response is received
+    And the response [has the 403 return code]
+    And the response has all other details as expected
+
+  @S-023
+  Scenario: must return negative response when request provides authentic credentials without authorised access
+    Given a user with [an active profile in CCD]
+    When a request is prepared with appropriate values
+    And the request [does not provide an authorised access to the operation]
+    And it is submitted to call the [Create event (V2)] operation of [CCD Data Store]
+    Then a negative response is received
+    And the response [has the 403 return code]
+    And the response has all other details as expected
+
+  @S-025
+  Scenario: must return negative response when request contains an invalid case-reference
+    Given a user with [an active profile in CCD]
+    When a request is prepared with appropriate values
+    And the request [contains an invalid case-reference]
+    And it is submitted to call the [Create event (V2)] operation of [CCD Data Store]
+    Then a negative response is received
+    And the response [has the 400 return code]
+    And the response has all the details as expected
+
+  @S-026 @Ignore #This scenario is returning 400 instead of expected 404, Need to raise defect JIRA
```

Comment:
The endpoint mentions the following for an invalid case ID.

    @Transactional
    @PostMapping(
        path = "/cases/{caseId}/events",
        headers = {
            V2.EXPERIMENTAL_HEADER
        },
        produces = {
            V2.MediaType.CREATE_EVENT
        }
    )
    @ApiOperation(
        value = "Submit event creation",
        notes = V2.EXPERIMENTAL_WARNING
    )
    @ApiResponses({
        @ApiResponse(
            code = 201,
            message = "Created",
            response = CaseResource.class
        ),
        **@ApiResponse(
            code = 400,
            message = V2.Error.CASE_ID_INVALID
        ),**
        @ApiResponse(
            code = 404,
            message = V2.Error.EVENT_TRIGGER_NOT_FOUND
        ),
        @ApiResponse(
            code = 409,
            message = V2.Error.CASE_ALTERED
        )
    })
Can you please discuss with a QA for proper response code? In my opinion, it should not be a bug but a QA can clarify the things.

Reply:
There are 2 scenarios, one being invalid case reference which returns 400 and already tested as @S-025. @S-026 refers the case reference structurally valid but non existent, in this case endpoint must return 404 not found but it is returning bad request which is incorrect.